docs/how-to/package-an-application.md

```diff
@@ -0,0 +1,138 @@
+
+### Declaring Textual as a dependency
+
+Your deployed application will depend on Textual, so be sure to tell the packaging tool that this is the case. This way, when an end user installs your application with a tool like [`pip`](https://pip.pypa.io/en/stable/) or [`pipx`](https://pipx.pypa.io/stable/), Textual will also be installed.
+The method used to add Textual as a dependency will differ depending on the tool you use, but the result will be an entry being added to `pyproject.toml`:
+
+=== "pyproject.toml (Hatch)"
+
+    ```toml
+    [project]
+    ...
+    dependencies = [
+      "textual",
+    ]
+    ```
+
+=== "pyproject.toml (Poetry)"
+
+    ```toml
+    [tool.poetry.dependencies]
+    ...
+    textual = "*"
+    ```
+
+!!! note
+
+    You may wish to pin the version of Textual; perhaps to a minimum version, or even to the exact version you are using at that time.
+    Opinions on best practice here do vary.
+
+### Application entry point and the runnable command
+
+We're going to turn the Textual [calculator example](https://github.com/Textualize/textual/blob/main/examples/calculator.py) into an application that can be packaged, deployed to [PyPi](https://pypi.org/), and installed by users using [`pip`](https://pip.pypa.io/en/stable/) or [`pipx`](https://pipx.pypa.io/stable/).
+
+The end result we want is that, after the user has installed the package, they can type `calculator` in the shell and the application will appear.
+
+#### Making the entry point
+
+Normally, somewhere in a Textual application, you'll have some code that looks like this:
+
+```python
+if __name__ == "__main__":
+    CalculatorApp().run()
+```
+
+We're going to make a small change to this to help us when it comes to telling the packaging tool how to run the application.
+The change is to turn the above into this:
+
+```python
+def run() -> None:
+    """Run the calculator application."""
+    CalculatorApp().run()
+
+if __name__ == "__main__":
+    run()
+```
+
+In other words:
+we've created a function called `run` that runs the calculator application;
+then we've changed the `__main__` test to run that function (we keep that because it lets us run our application with `python -m`; this can be useful during development).
+
+#### Declaring the runnable command
+
+Having done the above, we're all set to tell the packaging tool what to call the final command that the user will run, and what it should do when they run it.
+For Hatch and Poetry it is a case of adding:
+
+```
+<application-name> = "<entry-point>"
+```
+
+to the correct section of `pyproject.toml`.
+In the case of our calculator example, the command we want is `calculator` and the entry point is the `run` function in the module `calculator.py` which is in the package `textual_calculator`;
+we specify this with `textual_calculator.calculator:run`.
```

Comment:
I think we should explain this syntax. i.e. that it has the import path, a colon, and the function name.